### ISSUE TITLE
`$rolling()` method with temporal window ignores `min_periods`

### ISSUE THREAD

**Issue by fdetsch** [pola-rs/r-polars#546]
I was playing around with the `$rolling()` method and time windows as newly introduced in package version `0.11.0`. In this context, it seems like `min_periods` is not respected when calculating the rolling quantile with a temporal moving window (also tested with rolling max). 

Using the sample code from below, I would expect columns 

* `"roll_quant_int"` that is calculated based on a fixed-width integer window and 
* `"roll_quant_time"` that implements a temporal moving window

to be identical. However, the latter approach imputes values at the first two positions despite the `min_periods = 3L` requirement not being fulfilled. Am I missing something here or does this still need to be resolved?

```r
library(polars)

## sample data
dat = transform(
  airquality
  , "Date" = as.Date(
    paste(
      "1973"
      , Month
      , Day
      , sep = "-"
    )
  )
) |> 
  as_polars_df()

## set sorted flag
dat = dat$
  with_columns(
    pl$col("Date")$set_sorted()
  )

## compute rolling quantile ..
dat$
  with_columns(
    # .. with fixed integer window
    roll_quant_int = pl$
      col("Temp")$
      rolling_quantile(
        0.1
        , window_size = 5L
        , min_periods = 3L
        , by = "Date"
        , closed = "both"
      )
    # .. with dynamic temporal window
    , roll_quant_time = pl$
      col("Temp")$
      rolling_quantile(
        0.1
        , window_size = "4d"
        , min_periods = 3L
        , by = "Date"
        , closed = "both"
      )
  )

shape: (153, 9)
┌───────┬─────────┬──────┬──────┬───┬─────┬────────────┬────────────────┬─────────────────┐
│ Ozone ┆ Solar.R ┆ Wind ┆ Temp ┆ … ┆ Day ┆ Date       ┆ roll_quant_int ┆ roll_quant_time │
│ ---   ┆ ---     ┆ ---  ┆ ---  ┆   ┆ --- ┆ ---        ┆ ---            ┆ ---             │
│ i32   ┆ i32     ┆ f64  ┆ i32  ┆   ┆ i32 ┆ date       ┆ f64            ┆ f64             │
╞═══════╪═════════╪══════╪══════╪═══╪═════╪════════════╪════════════════╪═════════════════╡
│ 41    ┆ 190     ┆ 7.4  ┆ 67   ┆ … ┆ 1   ┆ 1973-05-01 ┆ null           ┆ 67.0            │
│ 36    ┆ 118     ┆ 8.0  ┆ 72   ┆ … ┆ 2   ┆ 1973-05-02 ┆ null           ┆ 67.0            │
│ 12    ┆ 149     ┆ 12.6 ┆ 74   ┆ … ┆ 3   ┆ 1973-05-03 ┆ 67.0           ┆ 67.0            │
│ 18    ┆ 313     ┆ 11.5 ┆ 62   ┆ … ┆ 4   ┆ 1973-05-04 ┆ 62.0           ┆ 62.0            │
│ …     ┆ …       ┆ …    ┆ …    ┆ … ┆ …   ┆ …          ┆ …              ┆ …               │
│ null  ┆ 145     ┆ 13.2 ┆ 77   ┆ … ┆ 27  ┆ 1973-09-27 ┆ 63.0           ┆ 63.0            │
│ 14    ┆ 191     ┆ 14.3 ┆ 75   ┆ … ┆ 28  ┆ 1973-09-28 ┆ 63.0           ┆ 63.0            │
│ 18    ┆ 131     ┆ 8.0  ┆ 76   ┆ … ┆ 29  ┆ 1973-09-29 ┆ 63.0           ┆ 63.0            │
│ 20    ┆ 223     ┆ 11.5 ┆ 68   ┆ … ┆ 30  ┆ 1973-09-30 ┆ 68.0           ┆ 68.0            │
└───────┴─────────┴──────┴──────┴───┴─────┴────────────┴────────────────┴─────────────────┘
```

**Comment by fdetsch** [comment]
Good catch, I hadn't tested this in Python. Of course, feel free to transfer this issue as necessary. Thanks!

**Comment by etiennebacher** [comment]
https://github.com/pola-rs/polars/issues/12715

**Comment by etiennebacher** [comment]
This is now fixed in 0.12.1, which includes the related fix in upstream rust-polars

**Comment by fdetsch** [comment]
Confirmed, thanks!
